api/PayrollProcessor.Core.Domain/Intrastructure/Clocks/DateTimeProvider.cs

```diff
@@ -0,0 +1,17 @@
+using System;
+
+namespace PayrollProcessor.Core.Domain.Intrastructure.Clocks
+{
+    public interface IDateTimeProvider
```

Comment:
I didn't feel like creating separate files for these Interface abstractions that are so closely tied to their single implementations.